warp10/src/main/java/io/warp10/script/ext/http/HTTP.java

```diff
@@ -0,0 +1,449 @@
+    AtomicLong urlCount;
+    AtomicLong downloadSize;
+
+    Object ufCount = stack.getAttribute(HttpWarpScriptExtension.ATTRIBUTE_HTTP_REQUESTS);
+    Object ufSize = stack.getAttribute(HttpWarpScriptExtension.ATTRIBUTE_HTTP_SIZE);
+
+    if (null == ufCount || null == ufSize) {
+      urlCount = new AtomicLong();
+      downloadSize = new AtomicLong();
+      stack.setAttribute(HttpWarpScriptExtension.ATTRIBUTE_HTTP_REQUESTS, urlCount);
+      stack.setAttribute(HttpWarpScriptExtension.ATTRIBUTE_HTTP_SIZE, downloadSize);
+    } else {
+      urlCount = (AtomicLong) ufCount;
+      downloadSize = (AtomicLong) ufSize;
+    }
+
+    if (urlCount.addAndGet(1) > maxrequests) {
+      throw new WarpScriptException(getName() + " is limited to " + maxrequests + " calls.");
+    }
+
+    Map<String, Object> res = new HashMap<>();
+    HttpURLConnection conn = null;
+
+    try {
+      conn = (HttpURLConnection) url.openConnection();
+
+      //
+      // Encode userinfo and set headers
+      //
+
+      if (null != authInfo) {
+
+        Map additionalHeaders;
+        if (null != authMacro) {
+          stack.push(authInfo);
+          stack.exec(authMacro);
+          additionalHeaders = (Map) stack.pop();
+
+        } else {
+          // doing basic auth
+          if (authInfo.size() != 2) {
+            throw new WarpScriptException(getName() + " expects a list with two items, username and password, in argument " + authInfo + ".");
+          }
+
+          if (!(authInfo.get(0) instanceof String)) {
+            throw new WarpScriptException(getName() + " expects a STRING username when using basic authentication.");
+          }
+
+          if (!(authInfo.get(1) instanceof String)) {
+            throw new WarpScriptException(getName() + " expects a STRING password when using basic authentication.");
+          }
+
+          String userInfo = authInfo.get(0) + ":" + authInfo.get(1);
+          String basicAuth = "Basic " + Base64.encodeBase64String(userInfo.getBytes(StandardCharsets.UTF_8));
+          additionalHeaders =  new HashMap<Object, Object>();
+          additionalHeaders.put("Authorization", basicAuth);
+        }
+
+        headers.putAll(additionalHeaders);
+      }
+
+      for (Map.Entry<Object, Object> prop: headers.entrySet()) {
+        conn.setRequestProperty(String.valueOf(prop.getKey()), String.valueOf(prop.getValue()));
+      }
+
+      conn.setDoInput(true);
+      conn.setRequestMethod(method.toUpperCase());
+
+      //
+      // Set body
+      //
+
+      if (body instanceof String) {
+        String bodyS = (String) body;
+        conn.setDoOutput(bodyS.length() > 0);
+        if (bodyS.length() > 0) {
+          try (OutputStream os = conn.getOutputStream()) {
+            os.write(bodyS.getBytes(StandardCharsets.UTF_8));
+          }
+        }
+
+      } else if (body instanceof byte[]) {
+        byte[] bodyB = (byte[]) body;
+        conn.setDoOutput(bodyB.length > 0);
```

Comment:
ditto, body should be ignored if method is GET or another method which does not allow a body